tests/event/listener_test.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ *
+ * Collapsible Categories extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2015 phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\collapsiblecategories\tests\event;
+
+class listener_test extends \phpbb_test_case
+{
+	/** @var \phpbb\collapsiblecategories\event\listener */
+	protected $listener;
+
+	/** @var \PHPUnit_Framework_MockObject_MockObject */
+	protected $controller_helper;
+
+	/** @var \PHPUnit_Framework_MockObject_MockObject */
+	protected $operator;
+
+	/** @var \PHPUnit_Framework_MockObject_MockObject */
+	protected $template;
+
+	/**
+	 * Setup test environment
+	 */
+	public function setUp()
+	{
+		parent::setUp();
+
+		// Stub of the operator class
+		$this->operator = $this->getMockBuilder('\phpbb\collapsiblecategories\operator\operator_interface')
+			->getMock();
+
+		// Stub of the controller helper class
+		$this->controller_helper = $this->getMockBuilder('\phpbb\controller\helper')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->controller_helper->expects($this->any())
+			->method('route')
+			->willReturnCallback(function ($route, array $params = array()) {
+				return $route . '#' . serialize($params);
+			});
+
+		// Stub of the template class
+		$this->template = $this->getMockBuilder('\phpbb\template\template')
+			->getMock();
+	}
+
+	/**
+	 * Create our event listener
+	 */
+	protected function set_listener()
+	{
+		$this->listener = new \phpbb\collapsiblecategories\event\listener(
+			$this->controller_helper,
+			$this->operator,
+			$this->template
+		);
+	}
+
+	/**
+	 * Test the event listener is constructed correctly
+	 */
+	public function test_construct()
+	{
+		$this->set_listener();
+		$this->assertInstanceOf('\Symfony\Component\EventDispatcher\EventSubscriberInterface', $this->listener);
+	}
+
+	/**
+	 * Test the event listener is subscribing events
+	 */
+	public function test_getSubscribedEvents()
+	{
+		$this->assertEquals(array(
+			'core.display_forums_after',
+			'core.display_forums_modify_category_template_vars',
+		), array_keys(\phpbb\collapsiblecategories\event\listener::getSubscribedEvents()));
+	}
+
+	/**
+	 * Test setup_collapsible_categories() is assigning the expected template vars
+	 */
+	public function test_init_collapsible_categories()
+	{
+		$this->set_listener();
+
+		// Set up expectations
+		$this->template->expects($this->once())
+			->method('assign_vars')
+			->with(array(
+				'UA_COLLAPSIBLE_CATEGORIES_URL' => 'phpbb_collapsiblecategories_main_controller#' . serialize(array()),
```

Comment:
@rxu this is something different and only related to this test of the controller helper route test stub